test/e2e/ocp_loadbalancer.go

```diff
@@ -61,14 +61,15 @@ var _ = Describe("Load Balancer", Label(smoke), Ordered, func() {
 			ip = svc.Status.LoadBalancer.Ingress[0].IP
 		}, DefaultEventuallyTimeout, 10*time.Second, ctx).Should(Succeed())
 
-		By("checking the service is reachable and it returns a 200 OK")
+		By("checking the service is reachable and it returns a 404 NotFound")
 		Eventually(func(g Gomega, ctx context.Context) {
 			client := http.Client{Timeout: 10 * time.Second}
 			req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("http://%s", ip), nil)
 			g.Expect(err).NotTo(HaveOccurred())
 			resp, err := client.Do(req)
 			g.Expect(err).NotTo(HaveOccurred())
-			g.Expect(resp.StatusCode).To(Equal(http.StatusOK))
+			// Since we use mcr.microsoft.com/cbl-mariner/base/nginx image, 404 NotFound is expected.
+			g.Expect(resp.StatusCode).To(Equal(http.StatusNotFound))
 		}, DefaultEventuallyTimeout, 10*time.Second, ctx).Should(Succeed())
```

Comment:
Sorry if this is a stupid question (or if I am missing some context): does it make sense to test that we get an HTTP 404 when doing that HTTP call?  We could get a 404 for different reasons and maybe we are getting a 404 response but not due to the reason we think it is, even the test/assertion is "green".

Reply:
True! Also, what would happen is that _client.Do(req)_ would return an error and we would fail before checking the statuscode (and resp would be nil). 👍